--- origsrc/qupzilla/src/main/main.cpp	2012-07-22 00:46:19.000000000 -0500
+++ src/qupzilla/src/main/main.cpp	2012-07-22 00:54:51.911705500 -0500
@@ -20,7 +20,7 @@
 
 #include <QMessageBox> // For QT_REQUIRE_VERSION
 
-#ifdef Q_WS_X11
+#ifdef Q_OS_LINUX
 #include <iostream>
 #include <signal.h>
 #include <execinfo.h>
@@ -114,9 +114,11 @@ int main(int argc, char* argv[])
 #ifdef Q_WS_X11
     QApplication::setGraphicsSystem("raster"); // Better overall performance on X11
 
+#ifdef Q_OS_LINUX
     signal(SIGSEGV, qupzilla_signal_handler);
     signal(SIGPIPE, qupzilla_signal_handler);
 #endif
+#endif
 
     MainApplication app(argc, argv);
 
